exist-core/src/main/java/org/exist/webstart/JnlpServlet.java

```diff
@@ -59,9 +60,9 @@ public void init() throws ServletException {
 
         final Optional<Path> libDir = or(
             Optional.ofNullable(System.getProperty("app.repo")).map(Paths::get).filter(Files::exists),
-            () -> or (
+            (Supplier<Optional<Path>>) () -> or (
```

Comment:
Please add an explanatory comment why we need that here as discussed in the community call.

Reply:
@line-o  Do we have some special prefix in order to find such place for later clean up in case this problem gets fixed?